### PR TITLE
8280400: JDK 19 L10n resource files update - msgdrop 10

### PR DESCRIPTION
**Pull request by alisenchung**
msg drop for jdk19, Mar 9, 2022

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8280400](https://bugs.openjdk.java.net/browse/JDK-8280400): JDK 19 L10n resource files update - msgdrop 10

### Reviewers
 * [Naoto Sato](https://openjdk.java.net/census#naoto) (@naotoj - **Reviewer**)
 * [Alexander Zuev](https://openjdk.java.net/census#kizune) (@azuev-java - **Reviewer**) ⚠️ Review applies to 4735722d75df87960d546c0bda72b072cf65a274

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7765/head:pull/7765` \
`$ git checkout pull/7765`

Update a local copy of the PR: \
`$ git checkout pull/7765` \
`$ git pull https://git.openjdk.java.net/jdk pull/7765/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7765`

View PR using the GUI difftool: \
`$ git pr show -t 7765`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7765.diff">https://git.openjdk.java.net/jdk/pull/7765.diff</a>

</details>

### CODE REVIEW

**Review by naotoj — COMMENTED** [review]
`src/java.base/share/classes/sun/util/resources/CurrencyNames_de.properties`
`src/java.base/share/classes/sun/util/resources/CurrencyNames_ja.properties`
`src/java.base/share/classes/sun/util/resources/CurrencyNames_zh_CN.properties`

These are part of `jdk.localedata` module. Should be excluded from `java.base` module and apply the diffs to `src/jdk.localedata/share/classes/sun/util/resources/ext/CurrencyNames_xx.properties` manually. (Note that the first half of it is not necessary when merging).

**Comment by JoeWang-Java** [comment]
For the bundles in java.xml:

For files with Oracle copyright, update the year to 2022 and @LastModified Mar 2022. Take XPATHErrorResources_ja.java as an example, the copyright year was updated to 2021 and @LastModified: Nov 2021. That's probably the date it was last modified, but as we updating them now along with a number of other files, it would be good to be consistent and change all of them to the current date.

For files with the reserved comment block such as SerializerMessages_de.java, do the same as did in XPATHErrorResources_de.java, add the copyright header and LastModified tag with the current date. 

For files with the Oracle GPL license such as message_zh_CN.properties, do not delete the license header. Instead, update the copyright year to 2022 if there are changes. I don't see any changes were made for many of these files, for example from msg/XMLSchemaMessages_ja.properties to regex/message_zh_CN.properties, the only change was the removal of the header.

In the webrev view, some files have the word "undefined" right under the license header, hopefully once the license header is fixed, that would go away as well.

**Comment by dfuch** [comment]
For simple webserver resource files - should the copyright year be 2022?

**Comment by naotoj** [comment]
IIRC, localized resource files should have the same copyright year as the base English one. That's what I was told by the 10n engineer when I had the same comment.

**Comment by dfuch** [comment]
> IIRC, localized resource files should have the same copyright year as the base English one. That's what I was told by the l10n engineer when I had the same comment.

Thanks Naoto! I have no objection then.

File: src/jdk.jdi/share/classes/com/sun/tools/example/debug/tty/TTYResources_de.java [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+        {"Thread description name sleepingStatus BP", "  {0} {1} inaktiv (bei Breakpoint)"},
+        {"Thread description name sleepingStatus",    "  {0} {1} inaktiv"},
+        {"Thread description name waitingStatus BP",  "  {0} {1} wartet in einem Monitor (bei Breakpoint)"},
+        {"Thread description name waitingStatus",     "  {0} {1} wartet in einem Monitor"},
+        {"Thread description name condWaitstatus BP", "  {0} {1} wartet bedingt (bei Breakpoint)"},
+        {"Thread description name condWaitstatus",    "  {0} {1} wartet bedingt"},
+        {"Thread has been resumed", "Thread wurde wiederaufgenommen"},
+        {"Thread not suspended", "Thread nicht unterbrochen"},
+        {"thread group number description name", "{0,number,integer}. {1} {2}"},
+        {"Threadgroup name not specified.", "Name der Threadgruppe nicht angegeben."},
```
Comment (plummercj) [inline src/jdk.jdi/share/classes/com/sun/tools/example/debug/tty/TTYResources_de.java:305]: Same comment here as mentioned in TTYResources_ja.java.

File: src/jdk.jdi/share/classes/com/sun/tools/example/debug/tty/TTYResources_de.java [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
...ck>          -- Verhindert die Garbage Collection eines Objekts\nenablegc <Ausdruck>          -- L\u00E4sst die Garbage Collection eines Objekts zu\n\n!!                        -- Wiederholt den letzten Befehl\n<n> <Befehl>             -- Wiederholt einen Befehl n-mal\nrepeat                    -- Zeigt an, ob die Wiederholung durch leeren Befehl im GDB-Stil aktiviert ist\nrepeat <on|off>           -- Aktiviert/deaktiviert die Wiederholung im GDB-Stil\n# <Befehl>               -- Verwerfen (kein Vorgang)\nhelp (oder ?)               -- Listet Befehle auf\ndbgtrace [flag]           -- Identisch mit der Befehlszeilenoption \"dbgtrace\"\nversion                   -- Gibt Versionsinformationen aus\nexit (oder quit)            -- Beendet den Debugger\n\n<Klassen-ID>: Ein vollst\u00E4ndiger Klassenname mit Package-Qualifiers\n<Klassenmuster>: Ein Klassenname mit einem Platzhalter am Anfang oder Ende (\"*\")\n<Thread-ID>: Threadnummer aus dem Befehl \"threads\"\n<Ausdruck>: Ein Ausdruck der Java(TM)-Programmiersprache.\nDer Gro\u00DFteil der g\u00E4ngigen Syntax wird unterst\u00FCtzt.\n\nStartbefehle k\u00F6nnen in \"jdb.ini\" oder \".jdbrc\" abgelegt werden\nin user.home oder user.dir"},
```
Comment (plummercj) [inline src/jdk.jdi/share/classes/com/sun/tools/example/debug/tty/TTYResources_de.java:342]: Same comment here as mentioned in TTYResources_ja.java.

File: src/jdk.localedata/share/classes/sun/util/resources/ext/CurrencyNames_de.properties [hunk 3]
```diff
@@ -60,218 +60,450 @@
 # or other dealings in these Data Files or Software without prior
 # written authorization of the copyright holder.
 
-#
-# Generated automatically from the Common Locale Data Repository. DO NOT EDIT!
-#
+ADP=ADP
```
Comment (naotoj) [inline src/jdk.localedata/share/classes/sun/util/resources/ext/CurrencyNames_de.properties:63]: No need for these lines in each localized files, as English bundle contains them.

File: src/jdk.localedata/share/classes/sun/util/resources/ext/CurrencyNames_ja.properties [hunk 4]
```diff
@@ -60,219 +60,450 @@
...
-uss=\u7c73\u30c9\u30eb (\u5f53\u65e5)
-uyu=\u30a6\u30eb\u30b0\u30a2\u30a4 \u30da\u30bd
-uzs=\u30a6\u30ba\u30d9\u30ad\u30b9\u30bf\u30f3 \u30b9\u30e0
-veb=\u30d9\u30cd\u30ba\u30a8\u30e9 \u30dc\u30ea\u30d0\u30eb
-vef=\u30d9\u30cd\u30ba\u30a8\u30e9 \u30dc\u30ea\u30d0\u30eb\u30d5\u30a8\u30eb\u30c6
-vnd=\u30d9\u30c8\u30ca\u30e0 \u30c9\u30f3
-vuv=\u30d0\u30cc\u30a2\u30c4 \u30d0\u30c4
-wst=\u897f\u30b5\u30e2\u30a2 \u30bf\u30e9
-xaf=CFA \u30d5\u30e9\u30f3 BEAC
+ADP=ADP
```
Comment (naotoj) [inline src/jdk.localedata/share/classes/sun/util/resources/ext/CurrencyNames_ja.properties:63]: Same here with `de` bundle.

File: src/jdk.localedata/share/classes/sun/util/resources/ext/CurrencyNames_zh_CN.properties [hunk 5]
```diff
@@ -60,220 +60,450 @@
...
-xpt=\u94c2
-xts=\u4e3a\u6d4b\u8bd5\u4fdd\u7559\u7684\u4ee3\u7801
-xxx=\u8d27\u5e01\u672a\u77e5\u6216\u65e0\u6548
-yer=\u4e5f\u95e8\u91cc\u4e9a\u5c14
-yum=\u5357\u65af\u62c9\u592b\u504c\u5a01\u7b2c\u7eb3\u5c14
-zar=\u5357\u975e\u5170\u7279
-zmk=\u8d5e\u6bd4\u4e9a\u514b\u74e6\u67e5
-zwd=\u6d25\u5df4\u5e03\u97e6\u5143
-zwl=\u6d25\u5df4\u5e03\u97e6\u5143 (2009)
+ADP=ADP
```
Comment (naotoj) [inline src/jdk.localedata/share/classes/sun/util/resources/ext/CurrencyNames_zh_CN.properties:63]: and here.

**Comment by wangweij** [comment]
The security related changes look fine, except for the year 2021.

File: src/jdk.jdi/share/classes/com/sun/tools/example/debug/tty/TTYResources_ja.java [hunk 6]
```diff
@@ -302,6 +302,7 @@
         {"Thread has been resumed", "\u30B9\u30EC\u30C3\u30C9\u304C\u518D\u958B\u3057\u307E\u3057\u305F"},
         {"Thread not suspended", "\u30B9\u30EC\u30C3\u30C9\u306F\u4E2D\u65AD\u3057\u3066\u3044\u307E\u305B\u3093"},
         {"thread group number description name", "{0,number,integer}. {1} {2}"},
+        {"Threadgroup name not specified.", "\u30B9\u30EC\u30C3\u30C9\u30B0\u30EB\u30FC\u30D7\u540D\u304C\u6307\u5B9A\u3055\u308C\u3066\u3044\u307E\u305B\u3093\u3002"},
```
[the remark this answers is not in the document]
Comment (alisenchung) [reply src/jdk.jdi/share/classes/com/sun/tools/example/debug/tty/TTYResources_ja.java:305]: The translation was started before the updates to this file. This update can be done in the next msg drop.

**Comment by azuev-java** [comment]
Clientlibs related stuff looks correct.

**Review by azuev-java — APPROVED** [review]

File: src/jdk.compiler/share/classes/com/sun/tools/javac/resources/ct_ja.properties [hunk 7]
```diff
@@ -0,0 +1,770 @@
+#
```
Comment (naotoj) [inline src/jdk.compiler/share/classes/com/sun/tools/javac/resources/ct_ja.properties:1]: The change to this file (moving ct.properties to ct_ja.properties) does look suspicious. Looks like this is not a translation file, but some kind of a configuration for `javac`. If so, it should not be translated (and translation configuration in the closed repository should exclude this file)

**Review by naotoj — APPROVED** [review]
LGTM. Thanks for the change.

**Pull request closed without merging**